include/picongpu/fields/MaxwellSolver/YeePML/YeePML.kernel

```diff
@@ -0,0 +1,587 @@
+                        auto const dBzDy = Dy( localB ).z( );
+
+                        /* Update convolutional fields using [Taflove, Hagness],
+                         * eq. (7.105a,b) and similar for other components.
+                         * For PIC the right-hand side uses B, not H.
+                         */
+                        auto & psiE = fieldPsiE( idx );
+                        psiE.yx = coeffs.b.x( ) * psiE.yx + coeffs.c.x( ) * dBzDx;
+                        psiE.zx = coeffs.b.x( ) * psiE.zx + coeffs.c.x( ) * dByDx;
+                        psiE.xy = coeffs.b.y( ) * psiE.xy + coeffs.c.y( ) * dBzDy;
+                        psiE.zy = coeffs.b.y( ) * psiE.zy + coeffs.c.y( ) * dBxDy;
+                        psiE.xz = coeffs.b.z( ) * psiE.xz + coeffs.c.z( ) * dByDz;
+                        psiE.yz = coeffs.b.z( ) * psiE.yz + coeffs.c.z( ) * dBxDz;
+
+                        /* [Taflove, Hagness], eq. (7.106) and similar for other
+                         * components. Coefficients Ca, Cb as given in (7.107a,b)
+                         * are general to account for materials, in addition to
+                         * artificial PML absorbing medium. We do not have any
+                         * real material, so in (7.107a,b) have to use
+                         * sigma(i + 1/2, j, k) = 0 (it is another sigma,
+                         * unrelated to PML), eps(i + 1/2, j, k) = EPS0. Also,
+                         * same as the Yee scheme in PIC, adjust to use B,
+                         * not H, in the right-hand side.
+                         */
+                        fieldE( idx ).x( ) += c2 * dt * (dBzDy / coeffs.kappa.y( ) -
+                            dByDz / coeffs.kappa.z( ) + psiE.xy - psiE.xz );
+                        fieldE( idx ).y( ) += c2 * dt * (dBxDz / coeffs.kappa.z( ) -
+                            dBzDx / coeffs.kappa.x( ) + psiE.yz - psiE.yx );
+                        fieldE( idx ).z( ) += c2 * dt * (dByDx / coeffs.kappa.x( ) -
+                            dBxDy / coeffs.kappa.y( ) + psiE.zx - psiE.zy );
+                    }
+                    else
+                        // Normal Yee scheme update
+                        fieldE( idx ) += curl( cachedB.shift( idxInSuperCell ) ) * c2 * dt;
+                }
+            );
+        }
+    };
+
+    /** Functor to update the magnetic field by half a time step
+     *
+     * @tparam T_numWorkers number of workers
+     * @tparam T_BlockDescription field (electric and magnetic) domain description
+     */
+    template<
+        uint32_t T_numWorkers,
+        typename T_BlockDescription
+    >
+    struct KernelUpdateBHalf
+    {
+        /** Update the magnetic field by half a time step
+         *
+         * @tparam T_Acc alpaka accelerator type
+         * @tparam T_Mapping mapper functor type
+         * @tparam T_Curl curl functor type
+         * @tparam T_EBox pmacc::DataBox, electric field box type
```

Comment:
I think the extra comma `,`is not needed here - same is true for the line below.

Reply:
Will fix.